src/docs/patterns/user-auth.mdx

```diff
@@ -0,0 +1,296 @@
+
+**Self-Registration**
+
+Self-registration involves registering an account without an email invitation. The workflow involves the following screens:
+
+-   Enter email address
+-   Accept EULA
+-   Verify email
+-   Create password
+-   Add account details
+-   Success & welcome
+
+**Registration via Invitation**
+
+Sometimes the organization admin might email the user an invitation. The workflow involves the following screens:
+
+-   Click through an email invitation
+-   Accept EULA
+-   Create password
+-   Add account details
+-   Success & welcome
+
+### Screens
+
+Most steps involved in self-registration and registration via invitation are very similar. Use the following screens for the registration workflow where appropriate.
+
+<ImageGrid
+    images={[EnterEmailAdress, AcceptEULA, VerifyEmail, CreatePassword, AccountDetails, WelcomeOnboard]}
+    caption={[
+        'Screens shared by the registration workflows.',
+        '1: Enter email address',
+        '2: Accept EULA',
+        '3: Verify email',
+        '4: Create password',
+        '5: Add account details',
+        '6: Success & welcome',
+    ]}
+/>
+
+## Joining an Organization
+
+In addition to account creation, some applications may require the user to join groups and organizations. Joining an organization workflow includes the following variations:
+
+**Request Access**
+
+This occurs when a user has an account but need to access an invitation-only organization. The workflow involves the following screens:
+
+-   Enter organization code
+-   Success & welcome (awaiting approval)
+-   Receive an email invitation
+-   Success & welcome
+
+**Enter Passcode**
+
+This occurs when anyone with an organization passcode can be added to the organization. The workflow involves the following screens:
+
+-   Enter organization code
+-   Enter organization passcode
+-   Success & welcome
+
+**Join with Invitation**
+
+An organization admin can add an existing user from their end too. In case the user doesn't have an account, they will first do the registration -> registration via invitation, then proceed to this step. The workflow involves the following screens:
+
+-   Receive an email invitation
+-   Success & welcome
+
+### Screens
+
+Most steps involved in the joining-an-organization workflow are very similar. Use the following screens for the joining-an-organization workflow where appropriate.
+
+<ImageGrid
+    images={[EnterOrganizationCode, EnterOrganizationPasscode, EnterOrganizationConfirm, EnterOrganizationWelcome]}
+    caption={[
+        'Screens shared by the joining-an-organization workflows.',
+        '1: Enter organization code',
+        '2: Enter organization passcode',
+        '3: Success & welcome (awaiting approval)',
+        '4: Success & welcome',
+    ]}
+/>
+
+## Login
+
+Login includes the following variations:
+
+**Basic Login**
+
+A simple login presents fields for the users to enter their credentials and a button to submit them for verification.
```

Comment:
This is a section that I think could be fleshed out a bit. What are the extras? What does clicking them do? When do you show them?

Reply:
You mean the "extra functions" mentioned in the caption?

Reply:
Yeah, like the "Forgot Password" (arguably not an extra, but rather required), "Contact Support", "Sign Up".

I'm just thinking like an extra 2 sentences below the image that quickly says something to the effect of "You should have a Forgot password button that will initiate the Forgot password Workflow (see below)"